api/src/main/java/com/inrupt/client/HttpStatus.java

```diff
@@ -0,0 +1,80 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.inrupt.client;
+
+import java.util.Arrays;
+
+public final class HttpStatus {
+
+    public static final int BAD_REQUEST = 400;
+    public static final int UNAUTHORIZED = 401;
+    public static final int FORBIDDEN = 403;
+    public static final int NOT_FOUND = 404;
+    public static final int METHOD_NOT_ALLOWED = 405;
+    public static final int NOT_ACCEPTABLE = 406;
+    public static final int CONFLICT = 409;
+    public static final int GONE = 410;
+    public static final int PRECONDITION_FAILED = 412;
+    public static final int UNSUPPORTED_MEDIA_TYPE = 415;
+    public static final int TOO_MANY_REQUESTS = 429;
+    public static final int INTERNAL_SERVER_ERROR = 500;
+
+    enum StatusMessages {
+        BAD_REQUEST(HttpStatus.BAD_REQUEST, "Bad Request"),
+        UNAUTHORIZED(HttpStatus.UNAUTHORIZED, "Unauthorized"),
+        FORBIDDEN(HttpStatus.FORBIDDEN, "Forbidden"),
+        NOT_FOUND(HttpStatus.NOT_FOUND, "Not Found"),
+        METHOD_NOT_ALLOWED(HttpStatus.METHOD_NOT_ALLOWED, "Method Not Allowed"),
+        NOT_ACCEPTABLE(HttpStatus.NOT_ACCEPTABLE, "Not Acceptable"),
+        CONFLICT(HttpStatus.CONFLICT, "Conflict"),
+        GONE(HttpStatus.GONE, "Gone"),
+        PRECONDITION_FAILED(HttpStatus.PRECONDITION_FAILED, "Precondition Failed"),
+        UNSUPPORTED_MEDIA_TYPE(HttpStatus.UNSUPPORTED_MEDIA_TYPE, "Unsupported Media Type"),
+        TOO_MANY_REQUESTS(HttpStatus.TOO_MANY_REQUESTS, "Too Many Requests"),
+        INTERNAL_SERVER_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "Internal Server Error");
+
+        private final int code;
+        final String message;
+
+        StatusMessages(final int code, final String message) {
+            this.code = code;
+            this.message = message;
+        }
+
+        static String getStatusMessage(final int statusCode) {
+            return Arrays.stream(StatusMessages.values())
+                .filter(status -> status.code == statusCode)
+                .findFirst()
+                .map(knownStatus -> knownStatus.message)
+                .orElseGet(() -> {
+                    // If the status is unknown, default to 400 for client errors and 500 for server errors
+                    if (statusCode >= 400 && statusCode <= 499) {
+                        return BAD_REQUEST.message;
+                    }
+                    return INTERNAL_SERVER_ERROR.message;
+                });
+        }
+    }
```

Comment:
I don't like this logic.
While I think I understand why it would seem helpful, I disagree with it.

Other implementations of HTTP status classes that I know of choose either to return nulls or to throw when they encounter unknown statuses.

The default chosen here is very opinionated and I don't think that a library like this one, a fairly low-level HTTP client library, should make such decisions about responses from the server.

I might even argue the it is always the server who is responsible for the content of HTTP status codes and phrases, so our library should just honor the ones generated by the server.

[Another example of prior art](https://github.com/inrupt/enterprise-authorization/pull/801#discussion_r1562485606) is the JBoss RESTEasy implementation of JAX-RS (I think), that has a special HTTP status phrase 'Unknown Code' for some cases.

Reply:
Do we really need to do this status codes mapping at all or could you just pass the data on.
```
        BAD_REQUEST(HttpStatus.BAD_REQUEST, "Bad Request"),
        UNAUTHORIZED(HttpStatus.UNAUTHORIZED, "Unauthorized"),
        FORBIDDEN(HttpStatus.FORBIDDEN, "Forbidden"),
        ...
```

Unless you do all status codes there are going to be problems like this. Will your `HttpStatus` be used as API?

Reply:
> I might even argue the it is always the server who is responsible for the content of HTTP status codes and phrases, so our library should just honor the ones generated by the server.

I would have preferred that as well, however not all HTTP libraries expose the HTTP status message, in particular `java.net.http.HttpResponse` does not, so the `com.inrupt.client.ResponseInfo` interface implements the smaller common denominator and doesn't expose the status message returned by the server.

From there, I don't think it is possible to expose what the server returned. The current behavior is aligned with https://www.rfc-editor.org/rfc/rfc9110.html#name-status-codes: 

> HTTP status codes are extensible. A client is not required to understand the meaning of all registered status codes, though such understanding is obviously desirable. However, a client MUST understand the class of any status code, as indicated by the first digit, and treat an unrecognized status code as being equivalent to the x00 status code of that class.

Based on this, I would say it is more correct to default to 400 and 500 for client and server status message respectively rather than returning a unique "Unknown" status message.